packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -202,9 +226,205 @@ export default async function initSanity(
+        if (!existsSync(srcPath)) {
+          await fs
+            .mkdir(srcPath, {recursive: true})
+            .catch(() => debug('Error creating folder %s', srcPath))
+        }
+      }
+
+      const studioPath = await prompt.single({
+        type: 'input',
+        message: 'What url do you want to use for the embedded studio?',
+        default: '/studio',
+        validate(input) {
+          if (!input.startsWith('/')) {
+            return 'Must start with /'
+          }
+
+          if (input.endsWith('/')) {
+            return 'Must not end with /'
+          }
+
+          // a-Z, 0-9, -, _ and /
+          if (!/^[a-zA-Z0-9-_\\/]+$/.test(input)) {
+            return 'Must only contain a-Z, 0-9, -, _ and /'
+          }
+
+          return true
+        },
+      })
+
+      const embeddedStudioPath = path.join(
+        srcPath,
+        `${studioPath}/`,
+        useAppDir ? '[[..index]]/page.tsx' : '[[...index]].tsx'
+      )
+
+      await writeOrOverwrite(
+        embeddedStudioPath,
+        (useAppDir ? sanityStudioAppTemplate : sanityStudioPagesTemplate).replace(
+          ':configPath:',
+          new Array(embeddedStudioPath.slice(workDir.length).split('/').length - 1)
+            .join('../')
+            .concat('sanity.config')
+        )
+      )
+
+      const sanityConfigPath = path.join(workDir, 'sanity.config.'.concat(fileEnding))
+      await writeOrOverwrite(
+        sanityConfigPath,
+        sanityConfigTemplate
+          .replace(':route:', embeddedStudioPath.slice(workDir.length).replace('src/', ''))
+          .replace(':basePath:', studioPath)
+      )
+    }
+
+    const sanityCliPath = path.join(workDir, 'sanity.cli.'.concat(fileEnding))
+    await writeOrOverwrite(sanityCliPath, sanityCliTemplate)
+
+    // write sanity folder files
+    const recursivelyWriteFiles = async (
+      files: Record<string, string | Record<string, string>>,
+      folderPath?: string
+    ) => {
+      for (const [filePath, content] of Object.entries(files)) {
+        if (filePath.includes('.') && typeof content === 'string') {
+          await writeOrOverwrite(
+            path.join(workDir, 'sanity', folderPath || '', filePath.concat(fileEnding)),
+            content
+          )
+        } else {
+          await mkdirSync(path.join(workDir, 'sanity', filePath), {recursive: true})
+          if (typeof content === 'object') {
+            await recursivelyWriteFiles(content, filePath)
+          }
+        }
+      }
+    }
+
+    // ask what kind of schema setup the user wants
+    const templateToUse: 'blog' | 'clean' = await prompt.single({
+      message: 'Select project template to use',
+      type: 'list',
+      choices: [
+        {
+          value: 'clean',
+          name: 'Clean project with no predefined schemas',
+        },
+        {
+          value: 'blog',
+          name: 'Blog (schema)',
+        },
+      ],
+      default: 'clean',
+    })
```

Comment:
Same as https://github.com/sanity-io/sanity/pull/4439/files#r1180310081
With an added thought that you might be able to use the `templates` const.

Reply:
I think I was considering it, but ended up not doing it because the next init doesn't follow the template structure. I'll have a look again, though.